plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.reporting
+
+import org.jetbrains.kotlin.KtNodeTypes
+import org.jetbrains.kotlin.KtSourceElement
+import org.jetbrains.kotlin.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.diagnostics.SourceElementPositioningStrategies
+import org.jetbrains.kotlin.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.formver.ErrorStyle
+import org.jetbrains.kotlin.formver.PluginErrors
+import org.jetbrains.kotlin.formver.embeddings.SourceRole
+import org.jetbrains.kotlin.formver.viper.errors.*
+
+class VerifierErrorInterpreter {
+    private var alreadyReported = false
```

Comment:
This variable doesn't quite capture what it says it does: you only set it for original viper errors, not for all errors.